Support version 1.0.0 of Hashdiff

gabebw opened the pull request:
In version 1.0.0, the `hashdiff` gem changed its module name from `HashDiff` to `Hashdiff` to prevent name collision with the `hash_diff` gem (more information: https://github.com/liufengyun/hashdiff/issues/45).

The API did not change.

This commit updates fixture_builder to support the new version of hashdiff.

If you'd like, I could add some code that will seamlessly use `Hashdiff` or `HashDiff` depending on which one is available, and make this not a breaking change. Up to you!

gabebw commented on .ruby-version:
```diff
@@ -1 +1 @@
-ruby-2.2.5
+ruby-2.3.8
```
I changed the Ruby version to a version that Travis supports.

gabebw commented on .travis.yml:
```diff
@@ -1,6 +1,4 @@
 language: ruby
-rvm:
```
This is no longer necessary because Travis reads the version from `.ruby-version`.

gabebw commented on lib/fixture_builder/configuration.rb:
```diff
@@ -135,7 +135,7 @@ def rebuild_fixtures?
         return true
       elsif file_hashes_from_disk != file_hashes_from_config
         puts '=> rebuilding fixtures because one or more of the following files have changed (see http://www.rubydoc.info/gems/hashdiff for diff syntax):'
-        HashDiff.diff(file_hashes_from_disk, file_hashes_from_config).map {|diff| print '   '; p diff}
```
This is the main change in this PR.

thewoolleyman commented:
Hi,

Thanks for the PR!

Haven't looked at the details, but yes, I'd definitely prefer a backward-compatible non-breaking change.

gabebw commented:
@thewoolleyman That's totally understandable 😄 

I've added code that will use the `Hashdiff` constant if available (in newer versions of hashdiff) but will fall back to `HashDiff` for older versions of the hashdiff gem.

I noticed that tests weren't failing with old versions of Hashdiff, so I added a little test that exercised the code path that refers to hashdiff. The current PR passes with versions of hashdiff from 0.0.2 to 1.0.0.

gabebw commented on test/fixture_builder_test.rb:
```diff
@@ -83,4 +82,19 @@ def test_absolute_rails_fixtures_path
+
+  def test_rebuilding_due_to_differing_file_hashes
+    create_and_blow_away_old_db
+    force_fixture_generation_due_to_differing_file_hashes
```
I copied the existing `test_serialization` test above and just changed this line.

thewoolleyman commented:
@gabebw So, I bumped the ruby version on master (to one that comes by default on Travis).

I'll take a look at this PR, but first can you please:

1. remove your commit that bumps ruby
2. rebase against master
3. squash (`rebase -i ...`) all your commits down to a single clean, atomic commit.

Thanks,
-- Chad

gabebw commented:
@thewoolleyman Rebased and pushed!

thewoolleyman commented:
Merged, thanks for the help!
